x-pack/test/apm_api_integration/tests/throughput/index.ts

```diff
@@ -0,0 +1,208 @@
+      apmApiSupertest({
+        endpoint: `GET /api/apm/backends/{backendName}/charts/throughput`,
+        params: {
+          path: { backendName: 'elasticsearch' },
+          query: {
+            ...commonQuery,
+            kuery: '',
+          },
+        },
+      }),
+    ]);
+    const serviceApiThroughput = serviceAPIResponse.body.items[0].throughput;
+
+    const throughputChartApiMean = meanBy(serviceThroughputAPIResponse.body.currentPeriod, 'y');
+
+    const transactionsThroughputSum = sumBy(
+      transactionsDetailsAPIResponse.body.transactionGroups,
+      'throughput'
+    );
+
+    const backendThroughputChartMean = meanBy(
+      backendThroughputChartAPIResponse.body.currentTimeseries,
+      'y'
+    );
+
+    return {
+      serviceApiThroughput,
+      throughputChartApiMean,
+      transactionsThroughputSum,
+      serviceDependencies: serviceDependencyAPIResponse.body.serviceDependencies.map((item) => ({
+        backendName: (item.location as BackendNode).backendName,
+        throughputValue: item.currentStats.throughput.value,
+      })),
+      topDependencies: topBackendsAPIResponse.body.backends.map((item) => ({
+        backendName: (item.location as BackendNode).backendName,
+        throughputValue: item.currentStats.throughput.value,
+      })),
+      backendThroughputChartMean,
+    };
+  }
+
+  let throughputValues: PromiseReturnType<typeof getThroughputValues>;
+
+  registry.when(
+    'Throughput value across apis',
+    { config: 'basic', archives: [archiveName] },
+    () => {
+      describe('with kql filter to force transaction-based UI', () => {
+        before(async () => {
+          throughputValues = await getThroughputValues({
+            serviceName: 'opbeans-go',
+            processorEvent: 'transaction',
+          });
+        });
+
+        it('matches throughput value between service, throughput chart and transactions apis ', () => {
+          const {
+            serviceApiThroughput,
+            throughputChartApiMean,
+            transactionsThroughputSum,
+          } = throughputValues;
+          expect(roundNumber(serviceApiThroughput)).to.be(roundNumber(throughputChartApiMean));
+          expect(roundNumber(serviceApiThroughput)).to.be(roundNumber(transactionsThroughputSum));
+        });
+
+        it('matches throughput value between service dependencies and top backends', () => {
+          const { topDependencies, serviceDependencies } = throughputValues;
+          expect(serviceDependencies).to.eql(topDependencies);
+        });
+
+        it('matches throughput value between service dependencies and backend throughput chart', () => {
+          const { backendThroughputChartMean, serviceDependencies } = throughputValues;
+          expect(roundNumber(serviceDependencies[0].throughputValue)).to.equal(
+            roundNumber(backendThroughputChartMean)
+          );
+        });
+      });
+
+      describe('with kql filter to force metric-based UI', () => {
```

Comment:
Instead of testing transaction-based and metric-based throughput separately would it make sense to compare them against each other for all of the APIs? Afaict there's only one APi where you compare them